internal/executors/http/http_executor.go

```diff
@@ -204,49 +205,35 @@ func execute(c http.Client, p *HttpRequestParameters, authHeader AuthorizationHe
 }
 
 func requestTimedOut(err error) bool {
-	if err == nil {
-		return false
-	}
-
 	var e net.Error
-	if errors.As(err, &e); e.Timeout() {
+	if isNetErr := errors.As(err, &e); err != nil && isNetErr && e.Timeout() {
```

Comment:
errors.As already returns false if `err == nil` so the `err != nil` check is not necessary.
```
func As(err error, target any) bool {
	if err == nil {
		return false
	}
```